src/main/java/me/xmrvizzy/skyblocker/skyblock/BackpackPreview.java

```diff
@@ -0,0 +1,192 @@
+package me.xmrvizzy.skyblocker.skyblock;
+
+import com.mojang.blaze3d.systems.RenderSystem;
+import me.xmrvizzy.skyblocker.SkyblockerMod;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.font.TextRenderer;
+import net.minecraft.client.gui.DrawableHelper;
+import net.minecraft.client.gui.screen.Screen;
+import net.minecraft.client.gui.screen.ingame.HandledScreen;
+import net.minecraft.client.render.item.ItemRenderer;
+import net.minecraft.client.util.math.MatrixStack;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.inventory.Inventory;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.*;
+import net.minecraft.util.Identifier;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class BackpackPreview extends DrawableHelper {
+    private static final Identifier TEXTURE = new Identifier(SkyblockerMod.NAMESPACE, "textures/gui/inventory_background.png");
+    private static final BackpackPreview instance = new BackpackPreview();
+    private static final Pattern ECHEST_PATTERN = Pattern.compile("Ender Chest.*\\((\\d+)/\\d+\\)");
+    private static final Pattern BACKPACK_PATTERN = Pattern.compile("Backpack.*\\((\\d+)/\\d+\\)");
+
+    private static final Inventory[] storage = new Inventory[27];
+    private static final boolean[] dirty = new boolean[27];
+    private static boolean loaded = false;
+    private static int counter = 0;
+
+    private static File getSaveDir() {
+        String uuid = MinecraftClient.getInstance().player.getUuidAsString();
+        File dir = new File("./config/skyblocker/backpack-preview/" + uuid + "/");
```

Comment:
Instead of relying on minecraft being run from it's directory I'd recommend using either `runDirectory` on `MinecraftClient` or `getConfigDir` on `FabricLoader`.

Reply:
Fixed in PR #49